### ISSUE TITLE
python3 tools directory provides a standalone python environment, but pip3 from there does not work

### ISSUE THREAD

**Issue by jonathan-saunders-ge** [microsoft/vcpkg#18857]
**Describe the bug**
If I try to run `pip3.exe` from a copy of the generated **installed\x86-windows\tools\python3** directory it gets an error indicating it is looking for python.exe in the vcpkg **build** directory, rather than where I copied the python3 directory to. Running `python.exe -m pip` works.

**Environment**
- OS: Windows 10 (20H2)
- Compiler: Microsoft Visual Studio 2019 Version 16.10.2.

```
C:\dev\github-real\vcpkg>vcpkg version
Vcpkg package management program version 2021-05-05-e8977e69d9a3fb462d9ad42013d83a7682706659

See LICENSE.txt for license information.

C:\dev\github-real\vcpkg>vcpkg list
bzip2:x86-windows                                  1.0.8#1          bzip2 is a freely available, patent free, high-q...
expat:x86-windows                                  2.4.1            XML parser library written in C
libffi:x86-windows                                 3.3#9            Portable, high level programming interface to va...
liblzma:x86-windows                                5.2.5#2          Compression library with an API similar to that ...
openssl:x86-windows                                1.1.1k#5         OpenSSL is an open source project that provides ...
python3:x86-windows                                3.9.5#1          The Python programming language
sqlite3:x86-windows                                3.35.5           SQLite is a software library that implements a s...
zlib:x86-windows                                   1.2.11#10        A compression library
```

**To Reproduce**
1. `./vcpkg install python3`
2. Copy **.\installed\x86-windows\tools\python3** directory to **C:\temp**
3. Rename **vcpkg** folder to **vcpkg-hide**
4. Run `C:\temp\python3\Scripts\pip3.exe --version` or `C:\temp\python3\Scripts\pip3.9.exe --version`
5. See the following error message (where c:\dev\github-real\vcpkg is the vcpkg **build** directory)
```
PS C:\> C:\temp\python3\Scripts\pip3.exe --version
Fatal error in launcher: Unable to create process using
'"c:\dev\github-real\vcpkg\packages\python3_x86-windows\tools\python3\python.exe"  "C:\temp\python3\Scripts\pip3.exe" --version':
The system cannot find the file specified.
```

**Expected behavior**
If I run `C:\temp\python3\python.exe -m pip --version`, I get this output. This is what I would expect from `pip3.exe` and `pip3.9.exe`.
```
PS C:\> C:\temp\python3\python.exe -m pip --version
pip 21.1.1 from C:\temp\python3\lib\site-packages\pip (python 3.9)
```

**Failure logs** 
None.

**Additional context**
Looking in the pip3.exe, I can see the following text strings:
```
#!c:\dev\github-real\vcpkg\packages\python3_x86-windows\tools\python3\python.exe
...
__main__.py# -*- coding: utf-8 -*-
import re
import sys
from pip._internal.cli.main import main
if __name__ == '__main__':
...
```

**Comment by JackBoosY** [comment]
This path was written in the executable, don't know whether it's by design.

**Comment by JackBoosY** [comment]
cc @Hoikas

**Comment by Hoikas** [comment]
Unfortunately, I am unable to reproduce. I'm currently using the 3.9.6 bump from #18754 and this is my output after copying `D:\vcpkg\installed\x86-windows\tools\python3` to `D:\python3`:

```
PS D:\> D:\python3\Scripts\pip3.exe --version
pip 21.1.3 from d:\vcpkg\packages\python3_x86-windows\tools\python3\lib\site-packages\pip (python 3.9)
```

You haven't, perchance, deleted the vcpkg `packages` directory, have you? pip should really be looking into the installed directory for its scripts. but that will need some additional investigation. Also, do you really need to copy the python3 tool out of vcpkg? That's not really compatible with the intent of the interpreter provided by the python3 port.

**Comment by jonathan-saunders-ge** [comment]
Thank you for looking at this. I think that actually does reproduce it. It says it is running from **d:\vcpkg\packages\python3_x86-windows\tools\python3** instead of from **D:\python3**.

Try renaming **d:\vcpkg** to **d:\vcpkg-hide**. When you run `D:\python3\Scripts\pip3.exe --version` you will see it tries to run from **d:\vcpkg** and will fail.

I was hoping that we can build python3 with vcpkg and then distribute it as a python runtime with our product. (I thought I had seen an issue or comment about someone doing that with a game they were making...) Perhaps that is not the intent of the port, though.

Is that python3 only meant to be a build tool used while building other vcpkg ports, or is it meant to be a redistributable?

So far, everything seems to be working great. As a work-around, we could just delete the pip3.exe and pip3.9.exe files and have people use `python.exe -m pip` instead of `pip.exe`, but it would be nice for them to be able to use `pip.exe`.

**Comment by Hoikas** [comment]
The intent for the python.exe built with vcpkg is to be used as part of a build process in general (not necessarily in vcpkg). I help maintain an open source game that uses Python, but we do not distribute the python.exe to end users; however, we do use packages installed from pip as part of our build process. Still, it would be good for this case to not error. Somehow the Python installer creates working pip.exe files, but I don't know how that works off the top of my head.

**Comment by jonathan-saunders-ge** [comment]
I can spend some time and take a look at it.

**Comment by jonathan-saunders-ge** [comment]
It looks like maybe **ensurepip** creates the pip3.exe. After I copied python3 someplace, I did this:
* deleted .\Scripts\pip*.exe,  .\Lib\site-packages\pip\, and  .\Lib\site-packages\pip-21.1.1.dist-info\
* ran `.\python.exe -m ensurepip`

And it recreated the pip*.exe files with the correct path embedded

So I guess this is not an issue with the port building at all. It is "post install" issue (which is beyond what the port needs to do).

Thanks for your help and sorry for taking up your time! (I should have dug more deeply before opening an issue.)
